### ISSUE TITLE
Memory corruption when using cloned images across threads

### ISSUE THREAD

**Issue by xaviershay** [dlemstra/Magick.NET#184]
The following script will indeterminately raise an `AccessViolationException` claiming corrupt memory:

```
IMagickImage image = new MagickImage("anyimage.png");
var lockObject = new Object();
var random = new Random();
Parallel.ForEach(Enumerable.Range(0, 1000), _ =>
{
    IMagickImage newImage = null;
    lock (lockObject)
    {
        newImage = image.Clone();
    }
    newImage.Resize(random.Next(100, 300), random.Next(100, 300));
    newImage.RePage();
    Console.WriteLine("{0}x{1}", newImage.Width, newImage.Height);
});
```

Expanding the lock to the `Resize` does not fix the issue.

Is there a supported way to get a "copy" of an image to work on on a different thread?

**Comment by dlemstra** [comment]
Do you have an NVIDIA video card? It might be fixed when you turn off OpenCL acceleration: `OpenCL.IsEnabled=false`. I cannot reproduce your issue on my machine.

p.s. Are you aware that the images are IDisposible? You need to put then in a using or call `.Dispose`.

**Comment by xaviershay** [comment]
> NVIDIA video card

Yes, a GTX1070. Your suggestion fixed the issue. Does this imply a bug in imagemagick itself?

> p.s. Are you aware that the images are IDisposible?

I was assuming that they were cleaned up properly on program exit. Guess I need to read up more on Disposable!

**Comment by dlemstra** [comment]
@xaviershay It is possible that a recent OpenCL bug that I found caused this issue. Would be very happy if you could give it a try after the next release.
